ansible/www-standalone/resources/config/nodejs.org

```diff
@@ -74,54 +95,98 @@ server {
+        # it attempts with /en/blog.html, which for the Website it will exist.
+        # This is basically a rewrite to remove the ".html" extension from our Website pages
+        # NOTE: By disabling trailingSlash config option on Next.js, less folders need to be created.
+        # If a file doesn't exist, it attempts to invoke the @english_fallback, as in most of cases
+        # for the Website, it means that, for example, /es/blog will not exist, but /en/blog exists
+        # so it attempts to open that page on its English version. Note that @english_fallback
+        # will only redirect two-letter-code pages to english ones, everything else goes right to 404.
         try_files $uri $uri.html @english_fallback;
 
         location ~ \.json$ {
             add_header access-control-allow-origin *;
         }
     }
 
-    # specific rule for handling next.js requests without overrides
-    location ^~ /static {
+    # This Location is used for handling static Next.js files. As we don't want to log access
+    # to static directories and also we don't want to log not found requests here
+    # As this is a static directory that in theory should not change over time, we disable access
+    # logs and also cache 404's errors as this folder contents change completely on every 
+    # We don't use ^~ as there are other Rewrite directives below that should also be taken into consideration
+    # before failing the request with a 404 if it doesn't exist
+    location /static {
         access_log off;
         log_not_found off;
-
-        try_files $uri =404;
+        open_file_cache_errors on;
     }
 
-    # specific rule for handling next.js requests without overrides
+    # This Location directy is used to handle Next.js internal _next directory
+    # As this is an internal directory requested by Next.js itself, we disable access
+    # logs and also cache 404's errors as this folder contents change completely on every build
+    # We use ^~ to tell NGINX to not process any other Location directive or Rewrite after this match
     location ^~ /_next {
         access_log off;
         log_not_found off;
-
-        try_files $uri =404;
+        open_file_cache_errors on;
     }
 
+    # This Location directive redirects to the list of all Node.js binaries available
+    # of all existing Node.js versions
+    # We use `open_file_cache_errors` that allows the NGINX file cache to also cache that
+    # certain files gave 404's, so that it doesn't keep retrying to open this file that doesn't exist
+    # This policy will still follow the expiry time of the file_cache. It is used mostly to avoid our file system
+    # to be hammered by concurrent requests and mitigate situations where the fs stops fulfiling requests
+    # because of maximum concurrent sockets being open.
+    # We use ^~ to tell NGINX to not process any other Location directive or Rewrite after this match
```

Comment:
```suggestion
    # This location directive exposes the dist directory, including index listings
    # This provides access to all Node.js binaries available for all Node.js versions
    # We use `open_file_cache_errors` here to allow NGINX to also cache metadata for missing files on disk
    # This still follows the expiry time of the open_file_cache. It is used mostly to avoid our file system
    # being hammered by concurrent requests and mitigate situations where the fs stops fulfiling requests
    # because of maximum concurrent sockets being open.
    # We use ^~ to tell NGINX to not process any other location directive or rewrite after this match
```
Same as above